cmd/label_list.go

```diff
@@ -83,6 +87,10 @@ func MapLabels(rn string, labelTerms []string) ([]string, error) {
 		if matches[i] == "" {
 			return nil, errors.Errorf("Label '%s' not found", term)
 		}
+
+		if ambiguous {
+			fmt.Printf("Label '%s' is ambiguous and may be a substring of another label\n", term)
```

Comment:
> just warn the user about it and keep the exact match.

That's not what the patch is doing, is it?

If there are substring matches for more than one label, then there is no exact match and the first match is returned. That feels quite arbitrary to me, in particular for actions like create/update, and especially for `label delete`.

Reply:
> If there are substring matches for more than one label, then there is no exact match and the first match is returned. 

Oh, of course, I *somehow* missed the `matches[i] != ""` there :/

> That feels quite arbitrary to me, in particular for actions like create/update, and especially for label delete

Not sure I follow your thought here.
The idea of returning the first "sub match" is the "arbitrary" part? If so, I completely agree!
Will fix it in v2.

Reply:
> The idea of returning the first "sub match" is the "arbitrary" part?

Yes. I'm thinking of cases like `lab label delete test` to delete a "test label" label, and end up deleting a "User Testing" label instead.